src/main/java/com/pusher/client/channel/impl/ChannelImpl.java

```diff
@@ -55,12 +57,13 @@ public void bind(final String eventName, final SubscriptionEventListener listene
         validateArguments(eventName, listener);
 
         Set<SubscriptionEventListener> listeners = eventNameToListenerMap.get(eventName);
```

Comment:
The synchronized block needs to include this read from the map:

- Thread 1 reads `eventNameToListenerMap`, gets `null`
- Thread 2 reads `eventNameToListenerMap`, gets `null`
- Thread 1 enters critical section, creates new HashSet containing listener 1 and stores in `eventNameToListenerMap`
- Thread 2 enters critical section, creates new HashSet containing listener 2 and stores in `eventNameToListenerMap`

Listener 1 has been lost.